### PR TITLE
Add some documentation for the allowed requires compare values

### PR DESCRIPTION
**Pull request by hughsie**

### CODE REVIEW

File: docs/source/metainfo.rst [hunk 1]
```diff
@@ -284,6 +284,28 @@ can be used.
...
++-----------+-------------+-----------------------+
+| ``le``    | ``1.2.3``   | Less than or equal    |
++-----------+-------------+-----------------------+
+| ``gt``    | ``1.2.3``   | Greater than          |
++-----------+-------------+-----------------------+
+| ``ge``    | ``1.2.3``   | Greater than or equal |
++-----------+-------------+-----------------------+
+| ``glob``  | ``??FWA*``  | Filename glob         |
++-----------+-------------+-----------------------+
+| ``regex`` | ``FW[1-7]`` | Regular expression    |
```
Comment (superm1) [inline docs/source/metainfo.rst]: for regex perhaps mention which regex format it you're using (BRE, ERE, SRE, perl)

Comment (hughsie) [reply docs/source/metainfo.rst]: I'm not sure myself! It's the one built into GLib itself, so "PCRE" maybe?

Comment (superm1) [reply docs/source/metainfo.rst]: Maybe link to this then: https://developer.gnome.org/glib/stable/glib-regex-syntax.html

Comment (hughsie) [reply docs/source/metainfo.rst]: I've clarified, thanks.

**Pull request merged**